matter_server/server/device_controller.py

```diff
@@ -1494,9 +1510,6 @@ def _on_mdns_operational_node_state(
 
         if not self._chip_device_controller.node_has_subscription(node_id):
             node_logger.info("Discovered on mDNS")
-        elif (now - last_seen) > NODE_MDNS_BACKOFF:
-            # node came back online after being offline for a while or restarted
-            node_logger.info("Re-discovered on mDNS")
```

Comment:
We need to properly test this - some magic things can happen if a device reboots within the backoff period or outside our mdns window and that results in a burst of removed and added events

Reply:
The removed events don't make it here, so we are good on these.

As for the Updated and Added events, we process fewer of those now: Only if there is no subscription (which is the same as before) we go into node_setup. So this certainly leads to less node_setup calls.

But it will more often call into the next branch, and call `trigger_resubscribe_if_scheduled`. But this doesn't do much other than setting a flag and forcing a timer timeout on SDK side.